jinahub/encoders/audio/AudioCLIPEncoder/audio_clip_encoder.py

```diff
@@ -17,25 +17,25 @@ class AudioCLIPEncoder(Executor):
     Encode audio data with AudioCLIP embeddings
     :param model_path: path of the pre-trained AudioCLIP model
     :param default_traversal_paths: default traversal path
+    :param device: Torch device string
     """
 
     TARGET_SAMPLE_RATE = 44100  # derived from ESResNeXt
 
     def __init__(
         self,
         model_path: str = 'assets/AudioCLIP-Full-Training.pt',
-        default_traversal_paths: Iterable[str] = ['r'],
+        default_traversal_paths: Iterable[str] = None,
+        device: str = 'cpu',
         *args,
         **kwargs
     ):
 
         super().__init__(*args, **kwargs)
         torch.set_grad_enabled(False)
         self.model_path = model_path
-        self.aclp = AudioCLIP(pretrained=model_path)
-        self.aclp.eval()
-        self.aclp.audio.eval()
-        self.default_traversal_paths = default_traversal_paths
+        self.aclp = AudioCLIP(pretrained=model_path).to(device).eval()
```

Comment:
```suggestion
        self.logger = JinaLogger(self.__class__.__name__)
        if device.startswith('cuda') and not torch.cuda.is_available():
            self.logger.warning(
                'You tried to use GPU but torch did not detect your'
                'GPU correctly. Defaulting to CPU. Check your CUDA installation!'
            )
            device = 'cpu'
        device = torch.device(device)
        self.aclp = AudioCLIP(pretrained=model_path).to(device).eval()
```

Reply:
We actually son't want a warning here, instead we want an error. We have removed all these warnings by choice. @tadejsv

Reply:
Yes, agreed. This warnings are hard to notice, unless you are constantly watching logs or have all the machinery around catching logs set up. If you specified that you want the executor to run on a GPU, but no GPU was found, there is a serious issue with the setup, and an error should be thrown, so the user is alerted immediately. It is still up to him whether to fix it, or leave it like that (by simply changing device to CPU)

We even missed these warnings when developing this, and thought our GPU executors were running fine. For users with less experience, this is even more likely

Reply:
that make sense